### DISCUSSION TITLE
Get active tab name in table list page and pass active tab name to create form

### DISCUSSION THREAD

**Discussion by elainecyx** [filamentphp/filament#13208] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.88

### How can we help you?

I define getTabs() function which return array of "plan category" in app\Filament\Resources\PlanResource\Pages\ListPlans class, then I want to customize create button label according to active tab, when user click create button, can show form modal with active tab that pass to form field named as "plan category",how should I do? 
screen captchure as below:
![image](https://github.com/filamentphp/filament/assets/79910670/d982577f-6df4-4e4b-9a68-a48cc6c34cb8)

I tried use "$this->activeTab" to get active tab name in ListPlans class, but it just get last active tab name, not the current one.
Thanks in advance!

**Comment by kulturman** [comment]
I am struggling with the same issue. Did you solve it?

**Reply by elainecyx** [reply]
not yet

**Comment by leandrocfe** [comment]
In your ListPage

```php
protected function getHeaderActions(): array
{
    return [
        Actions\CreateAction::make()
            ->label(fn (): string => 'label '.$this->activeTab)
            ->url(fn (): string => YourResource::getUrl('create', ['activeTab' => $this->activeTab])),
    ];
}
```

**Reply by leandrocfe** [reply]
@kulturman @elainecyx

**Reply by kulturman** [reply]
@leandrocfe With that I can't use a modal right? But anyway it's a good solution, thanks.

**Reply by leandrocfe** [reply]
> @leandrocfe With that I can't use a modal right? But anyway it's a good solution, thanks.

What would you do in the modal?

**Reply by kulturman** [reply]
> > @leandrocfe With that I can't use a modal right? But anyway it's a good solution, thanks.
> 
> What would you do in the modal?

Create the resource instead of going on "create" page.

**Reply by leandrocfe** [reply]
> > > @leandrocfe With that I can't use a modal right? But anyway it's a good solution, thanks.
> > 
> > What would you do in the modal?
> 
> Create the resource instead of going on "create" page.

Use a simple resource 

https://filamentphp.com/docs/3.x/panels/resources/getting-started/#simple-modal-resources

**Reply by elainecyx** [reply]
@leandrocfe Thanks for your help! I used the create form modal instead of create page because of business reason, so if use modal, how could I pass parameter to modal? or Can I get active tab in PlanResource?

**Reply by leandrocfe** [reply, marked as answer]
It depends. For example, if you want to fill a field value with the current tab, you could do this:

```php
protected function getHeaderActions(): array
{
    return [
        Actions\CreateAction::make()
            ->fillForm([
                'your_field' => $this->activeTab,
            ]),
    ];
}
```

**Reply by elainecyx** [reply]
Thanks for your solution !
